base/base.go

```diff
@@ -119,6 +132,9 @@ func (s *Server) Listen(handler Handler) error {
 	var eg errgroup.Group
 	eg.Go(func() error { return s.listenForMsgs(shutdownCh, sub, handler) })
 	eg.Go(func() error { return s.listenForConvs(shutdownCh, sub, handler) })
+	if s.multi != nil {
+		eg.Go(func() error { return s.multi.Heartbeat(shutdownCh) })
```

Comment:
can you check if nil inside of multi and just noop instead of having to check before each call?